addons/base_iban/models/res_partner_bank.py

```diff
@@ -28,6 +28,30 @@ def get_bban_from_iban(iban):
     """
     return normalize_iban(iban)[4:]
 
+def get_iban_part(iban, number_kind):
+    """ Get part of the iban depending on the mask
+        .. code-block:: python
+            # template = 'ITkk KBBB BBSS SSSC CCCC CCCC CCC'
+            partner.acc_number = 'IT60X0542811101000000123456'
+            get_iban_part(partner.acc_number, 'bank') == '05428'
+            get_iban_part(partner.acc_number,'account') == '000000123456'
+        Returns ``False`` in case of failure
+    """
+    kind_map = {
+        'bank': 'B',  # Bank national code
+        'branch': 'S',  # Branch code
+        'account': 'C',  # Account number
+        'check': 'k',  # Check digits
+        'check_other': 'K',  # Other check digits
```

Comment:
I would try to add the other Letters that are missing, like N/T/G/M (don't know if I missed some) if we're doing a helper.

Reply:
I'll try to find out what they are.